workbench-core/base/src/services/metadataService.ts

```diff
@@ -44,4 +63,124 @@ export class MetadataService {
       dependencyMetadata
     };
   }
+
+  /**
+   *
+   * @param mainEntityResourceType - main entity resource type
+   * @param mainEntityId - main entity id
+   * @param dependencyResourceType - dependency resource type
+   * @param parser - Zod parser for metadata object
+   * @param queryParams - parameters to query metadata. Supported page size and pagination token.
+   * @returns object containind list of metadata objects and continuation token.
+   */
+  public async listDependentMetadata<DependencyMetadata extends { pk: string; sk: string; id: string }>(
+    mainEntityResourceType: string,
+    mainEntityId: string,
+    dependencyResourceType: string,
+    parser: ZodTypeAny,
+    queryParams?: {
+      pageSize?: number;
+      paginationToken?: string;
+    }
+  ): Promise<{ data: DependencyMetadata[]; paginationToken: string | undefined }> {
+    let params: QueryParams = {
+      key: { name: 'pk', value: `${mainEntityResourceType}#${mainEntityId}` },
+      sortKey: 'sk',
+      begins: { S: `${dependencyResourceType}#` },
+      limit: Math.min(queryParams?.pageSize ?? DEFAULT_API_PAGE_SIZE, MAX_API_PAGE_SIZE)
+    };
+
+    params = addPaginationToken(queryParams?.paginationToken, params);
+
+    const result: QueryCommandOutput = await this._ddbService.query(params).execute();
+    if (!result?.Items) {
+      return { data: [], paginationToken: undefined };
+    }
+
+    const data: DependencyMetadata[] = [];
+    result.Items.forEach((item) => {
+      data.push(validateAndParse<DependencyMetadata>(parser, item));
+    });
+
+    return { data, paginationToken: getPaginationToken(result) };
+  }
+
+  public async updateRelationship<DependencyMetadata>(
+    mainEntityResourceType: string,
+    mainEntityId: string,
+    dependencyResourceType: string,
+    dependentIds: string[],
+    data: DependencyMetadata
+  ): Promise<void> {
+    if (!dependentIds.length) {
+      return;
+    }
+
+    if (dependentIds.length > MAX_BATCH_SIZE) {
+      throw new Error(`Cannot add more than ${MAX_BATCH_SIZE} dependencies in single batch.`);
```

Comment:
Thoughts on renaming `single batch` to `single transaction`? Batch and transactions have different meaning in a DDB context. However, I can swayed either way on whether we want to expose that concept in the `metaDataService`.